enhancements/cluster-logging/forward_to_azure_log_analytics.md

```diff
@@ -0,0 +1,182 @@
+        description: SharedKey the primary or the secondary key
+          for the Log Analytics workspace. https://learn.microsoft.com/en-us/azure/azure-monitor/logs/data-collector-api?tabs=powershell#authorization
+        type: string
+    type: object
+```
+
+
+### Workflow Description
+
+#### I want to forward logs to Azure Monitor: Log Analytics instead of a local store (common config)
+```yaml
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+  - name: AzureLogAnalytics
+    type: azureMonitor
+    azureMonitor:
+      customerId: "myCustomerID"
+      logType: "MyLogType"
+    secret:
+       name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+    outputRefs: [AzureLogAnalytics]
+```
+
+#### I want to forward logs to Azure Monitor: Log Analytics and associate the logs with some Azure resource
+```yaml
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+  - name: AzureLogAnalytics
+    type: azureMonitor
+    azureMonitor:
+      customerId: "myCustomerID"
+      logType: "MyLogType"
+      azureResourceId: "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/otherResourceGroup/examplestorage"
+    secret:
+       name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+    outputRefs: [AzureLogAnalytics]
+```
+
+#### I want to forward logs to Azure Monitor: Log Analytics in alternative host for dedicated Azure regions.
+```yaml
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+  - name: AzureLogAnalytics
+    type: azureMonitor
+    azureMonitor:
+      customerId: "myCustomerID"
+      logType: "MyLogType"
+      host: "ods.opinsights.azure.acme"
+    secret:
+       name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+    outputRefs: [AzureLogAnalytics]
+```
+
+### Implementation Details [optional]
+Use the [Vector plugin](https://vector.dev/docs/reference/configuration/sinks/azure_monitor_logs) to publish log events
+to the Azure Monitor Logs service
+Plugin configuration settings:
+
+The customer_id, log_type and shared_key fields are required, while the azure_resource_id and host field is optional.
+
+* customer_id : The unique identifier for the Log Analytics workspace.
+* log_type : Specify the record type of the data that's being submitted. It can contain only letters, numbers, and the underscore (_) character, and it can't exceed 100 characters. Each request to the Data Collector API must include a LogType with the name for the record type. The suffix _CL is automatically appended to the name you enter to distinguish it from other log types as a custom log. For example, if you enter the name MyNewRecordType, Azure Monitor creates a record with the type MyNewRecordType_CL. This helps ensure that there are no conflicts between user-created type names and those shipped in current or future Microsoft solutions.
```

Comment:
We should probably expand upon this.  I assume "log_type" is like a splunk or Elasticsearch index.  This means we can default them to our log types if not specified.  Additionally, I would expect that vector supports parameter expansion here meaning it could be dynamically set to a namespace.  Please investigate

Reply:
Unfortunately, `log_type` don't support template syntax. 
We can use some default value for sure, like `application`, `audit` and `infrastructure` as result in Azure we got 3 tables `application_CL`, `audit_CL` and `infrastructure_CL`